dask_planner/src/sql/optimizer/dynamic_partition_pruning.rs

```diff
@@ -0,0 +1,968 @@
+//! Optimizer rule for dynamic partition pruning (DPP)
+//!
+//! DPP refers to a query optimization rule in which distinct values in an inner join are used as
+//! filters in a table scan. This allows us to eliminate all other rows which do not fit the join
+//! condition from being read at all.
+//!
+//! Furthermore, a table involved in a join may be filtered during a scan, which allows us to
+//! further prune the values to be read.
+
+use std::{
+    collections::{HashMap, HashSet},
+    fs,
+    hash::{Hash, Hasher},
+};
+
+use datafusion_python::{
+    datafusion::parquet::{
+        basic::Type as BasicType,
+        file::reader::{FileReader, SerializedFileReader},
+        record::{reader::RowIter, RowAccessor},
+        schema::{parser::parse_message_type, types::Type},
+    },
+    datafusion_common::{Column, Result, ScalarValue},
+    datafusion_expr::{
+        logical_plan::LogicalPlan,
+        utils::from_plan,
+        Expr,
+        JoinType,
+        Operator,
+        TableScan,
+    },
+    datafusion_optimizer::{OptimizerConfig, OptimizerRule},
+};
+use log::warn;
+
+use crate::sql::table::DaskTableSource;
+
+// Optimizer rule for dynamic partition pruning
+pub struct DynamicPartitionPruning {}
+
+impl DynamicPartitionPruning {
+    #[allow(missing_docs)]
+    #[allow(dead_code)]
+    pub fn new() -> Self {
+        Self {}
+    }
+}
```

Comment:
Since this isn't used and the `DynamicPartitionPruning` struct has no members can we just remove this?